kernel/thread/thread.c

```diff
@@ -613,6 +627,20 @@ int thd_set_prio(kthread_t *thd, prio_t prio) {
 /*****************************************************************************/
 /* Scheduling routines */
 
+static void thd_update_cpu_time(kthread_t *thd) {
+    if(perf_cntr_timer_enabled()) {
+
+        const uint64_t ns = perf_cntr_timer_ns();
+
+        if(thd_current) {
```

Comment:
It just occurred to me. Is this intended to be just checking thd_current for NULL? Because that's basically an assertable condition and we don't check for it anywhere else. Technically we set it to NULL during initialization, so we have a check for it during thread creation just to have default pwds for kern/reaper/idle but after init it should always be set.